### PR TITLE
fix(tooltip): 修复自定义操作菜单传入自定义 ReactNode 不显示

### PR DESCRIPTION
**Pull request by lijinke666**
### 👀 PR includes

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [ ] Solve the issue and close #0

🔧 Chore

- [ ] Test case
- [ ] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 📝 Description

配置如下:

```ts
const s2Options: SheetComponentOptions = {
  tooltip: {
    operation: {
      menus: [
        {
          key: 'a',
          text: <span>文字</span>,
          icon: <StarOutlined />,
        },
      ],
    },
  }
}
```

1. **修复 `@antv/s2-react` 的 `SheetComponentOptions` 的类型** : 增加泛型, `core` 层 是 `Element | string` , `react` 层是 `React.ReactNode`

3. 修复 自定义操作菜单时, 传入自定义 ReactNode 的文本和 icon, tooltip 未能正常渲染: 原因是这里判断是非字符串, 直接取 `innerHTML`, 操作有点迷

image

### 🖼️ Screenshot

| Before | After |
| ------ | ----- |
| image  | image |
| image |  image |

### 🔍 Self-Check before the merge

- [ ] Add or update relevant docs.
- [ ] Add or update relevant demos.
- [ ] Add or update test case.
- [ ] Add or update relevant TypeScript definitions.

### 🖼️ Screenshot

| Before | After |
| ------ | ----- |
| image  | image |
| image |  image |

### 🔍 Self-Check before the merge

- [ ] Add or update relevant docs.
- [ ] Add or update relevant demos.
- [ ] Add or update test case.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: packages/s2-react/src/common/react-element.tsx [hunk 1]
```diff
@@ -1,27 +1,36 @@
 import React from 'react';
-import cx from 'classnames';
-import { S2_PREFIX_CLS, type TooltipContentType } from '@antv/s2';
+import cls from 'classnames';
```
Comment (lijinke666) [inline packages/s2-react/src/common/react-element.tsx:2]: 主要就改了这个文件, 其他全是透传泛型

File: packages/s2-react/src/components/tooltip/components/summary.tsx [hunk 2]
```diff
@@ -1,55 +1,57 @@
...
-    );
-  };
-
-  const renderSummary = () => {
-    return summaries?.map((item) => {
-      const { name = '', value } = item || {};
-      if (!name && !value) {
-        return;
-      }
+export const TooltipSummary: React.FC<TooltipSummaryProps> = React.memo(
```
Comment (lijinke666) [inline packages/s2-react/src/components/tooltip/components/summary.tsx:6]: 这个文件是格式化, 可忽略

**Review by stone-lyl — APPROVED** [review]
LG

**Pull request merged**